### PR TITLE
Rename anonymous to pure proxy

### PR DESCRIPTION
**Pull request by shawntabrizi**
Fixes: https://github.com/paritytech/substrate/issues/7735

Gav noted in the forum that he has been using the term "pure proxy", so this is a simple rename which changes that.

https://forum.polkadot.network/t/parachain-technical-summit-next-steps/51/14?u=shawntabrizi

This should have no runtime breaking changes, but APIs like Polkadot JS will probably need to do a rename to make this functionality work again, and also to update the branding of this kind of proxy account.

polkadot companion: https://github.com/paritytech/polkadot/pull/6038
cumulus companion: https://github.com/paritytech/cumulus/pull/1667

### CODE REVIEW

**Review by kianenigma — APPROVED** [review]
idea: It would have been awesome if somehow we could also generate a `#[deprecated]` section in the metadata. Then, tools like `PJS` api and subxt can still support the deprecated method name, but similar to `#[deprecated]` within rust, emit a warning that they should rename it to something else. 

Might be a big engineering mountain to climb initially, but it could potentially ease the development downstream a lot.

**Review by ggwpez — APPROVED** [review]

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/parachain-technical-summit-next-steps/51/20

**Comment by shawntabrizi** [comment]
cc @jacogr

**Comment by shawntabrizi** [comment]
bot merge

File: frame/proxy/src/lib.rs [hunk 1]
```diff
@@ -304,22 +288,17 @@ pub mod pallet {
...
 		/// Fails if there are insufficient funds to pay for deposit.
-		///
-		/// # <weight>
-		/// Weight is a function of the number of proxies the user has (P).
-		/// # </weight>
-		/// TODO: Might be over counting 1 read
-		#[pallet::weight(T::WeightInfo::anonymous(T::MaxProxies::get()))]
-		pub fn anonymous(
+		#[pallet::weight(T::WeightInfo::pure(T::MaxProxies::get()))]
+		pub fn pure(
```
Comment (joepetrowski) [inline frame/proxy/src/lib.rs:292]: Every other dispatchable function is a verb, for me at least it's a sensible pattern to follow.
```suggestion
		pub fn create_pure(
```

File: frame/proxy/src/lib.rs [hunk 2]
```diff
@@ -565,10 +516,10 @@ pub mod pallet {
 	pub enum Event<T: Config> {
 		/// A proxy was executed correctly, with the given.
 		ProxyExecuted { result: DispatchResult },
-		/// Anonymous account has been created by new proxy with given
+		/// pure account has been created by new proxy with given
```
Comment (joepetrowski) [inline frame/proxy/src/lib.rs:519]:
```suggestion
		/// A pure account has been created by new proxy with given
```

**Review by muharem — APPROVED** [review]

**Comment by shawntabrizi** [comment]
bot merge

**Comment by lucasvo** [comment]
I think this still causes major confusion because the word proxy is still used for two different things:

> Alice is a democracy proxy on the pure proxy Bob.

The number of people I've tried to explain the difference between Alice being a proxy and Bob being a different proxy is extremely confusing and I would suggest we call it `pure account` not `pure proxy`

**Comment by joepetrowski** [comment]
> Alice is a democracy proxy on the pure proxy Bob.

This doesn't make any sense though. What does pure account express? What makes an account created to access via proxy more pure than one accessed via cryptographic key or multisig?

**Comment by lucasvo** [comment]
@shawntabrizi don't you think calling this a proxy is a bad idea? Overloading the word `proxy` for both a permission on an account as well as an account itself makes this extremely awkward:

> Alice is a democracy proxy on the pure proxy Bob.

We use `pure proxies` as you call them quite extensively and no one in the team actually understands that the word proxy is used for two very different things.

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/parachain-technical-summit-next-steps/51/22

**Comment by lucasvo** [comment]
I apologize for my earlier comment, I think it's merely the wiki that is still not up to date on the new language.

**Pull request merged**